org.jacoco.core.test.validation.kotlin/src/org/jacoco/core/test/validation/kotlin/targets/KotlinInlineTarget.kt

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.validation.kotlin.targets
+
+import org.jacoco.core.test.validation.targets.Stubs.nop
+import org.jacoco.core.test.validation.targets.Stubs.t
+
+/**
+ * Test target for `inline` functions.
+ */
+object KotlinInlineTarget {
+
+    inline fun inlined() {
+        nop() // assertNotCovered()
+    }
+
+    @JvmStatic
+    fun main(args: Array<String>) {
+
+        inlined() // assertFullyCovered()
+
+        /* Following inlined method for some reasons doesn't appear in SMAP: */
+        assert(t()) // assertPartlyCovered(2, 2)
```

Comment:
Looks like intrinsics or whatever kick-in when using stdlib. When inlining from another source file it seems to work:

        KotlinInlineFun.inlined() // assertFullyCovered()

Or what is the point here?

Reply:
Thanks for clarification!